scripts/background/BadgeManager.js

```diff
@@ -2,31 +2,45 @@
 
 'use strict';
 
-function BadgeManager(addEventListener, isEnabled) {
-	let pausedState = {};
+function BadgeManager(addEventListener, isEnabledStart) {
+	let isEnabled = isEnabledStart;
+	let isTabAudible = false;
+	let badgeText;
+	let badgeIcon;
+
+	this.updateEnabled = (enabled) => {
+		printDebug("BadgeText has been set to", enabled);
+
+		isEnabled = enabled;
+
+		if (enabled) updateBadgeText();
+		else updateBadgeText(true);
+	}
 
 	addEventListener("hourMusic", (hour, weather) => {
-		isEnabled() && chrome.browserAction.setBadgeText({ text: `${formatHour(hour)}` });
+		badgeText = `${formatHour(hour)}`;
+		if (isEnabled) updateBadgeText();
 		setIcon(weather);
 	});
 
 	addEventListener("kkStart", () => {
-		isEnabled() && chrome.browserAction.setBadgeText({ text: "KK" });
+		badgeText = "KK";
+		if (isEnabled) updateBadgeText();
 		setIcon('kk');
 	});
 
 	addEventListener("pause", tabPause => {
-		chrome.browserAction.getBadgeText({}, badgeText => {
-			pausedState.badge = badgeText;
-			if (tabPause) chrome.browserAction.setBadgeText({ text: "ll" });
-			else chrome.browserAction.setBadgeText({ text: "" });
-			setIcon('paused');
-		});
+		if (tabPause) {
+			isTabAudible = true;
+			chrome.browserAction.setBadgeText({ text: "ll" });
```

Comment:
Didn't realise that this was the pause button 😂

Reply:
I mean, it looks pretty good when it runs lol

Reply:
The fact that I'm just noticing this now proves that is true